torch_geometric/nn/conv/gmn_conv.py

```diff
@@ -0,0 +1,150 @@
+            respective activation function defined by :obj:`act`.
+            (default: :obj:`None`)
+        train_norm (bool, optional): Whether normalization parameters
+            are trainable. (default: :obj:`False`)
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+
+    Shapes:
+        - **input:**
+          node features :math:`(|\mathcal{V}|, F_{in})`,
+          edge indices :math:`(2, |\mathcal{E}|)`,
+          edge features :math:`(|\mathcal{E}|, D)` *(optional)*
+        - **output:** node features :math:`(|\mathcal{V}|, F_{out})`
+    """
+    def __init__(self, in_channels: int, 
+                 out_channels: int,
+                 aggregators: List[str], 
+                 scalers: List[str], 
+                 deg: Tensor,
+                 edge_dim: Optional[int] = None, 
+                 towers: int = 1,
+                 post_layers: int = 1,  
+                 divide_input: bool = False, 
+                 act: Union[str, Callable, None] = "relu",
+                 act_kwargs: Optional[Dict[str, Any]] = None,
+                 **kwargs):
+
+        super().__init__(in_channels, out_channels, aggregators, scalers, deg, 
+                         edge_dim, towers, divide_input, **kwargs)
+
+        if divide_input:
+            assert in_channels % towers == 0
+        assert out_channels % towers == 0
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.aggregators = aggregators
+        self.scalers = scalers
+        self.edge_dim = edge_dim
+        self.towers = towers
+        self.divide_input = divide_input
+
+        self.F_in = in_channels // towers if divide_input else in_channels
+        self.F_out = self.out_channels // towers
+
+        if self.edge_dim is not None:
+            self.edge_encoder = Linear(edge_dim, self.F_in)
+
+        self.mix = ModuleList()
+        for _ in range(towers):
+
+            in_channels = (len(aggregators) * len(scalers) + 1) * self.F_in
+            modules = [Linear(in_channels, self.F_out)]
+
+            for _ in range(post_layers - 1):
+                x = self.F_out
+                modules += [nn.LayerNorm()]
+                modules += [torch.Transpose(1, 2)]
+                modules += [activation_resolver(act, **(act_kwargs or {}))]
+                modules += [Linear(self.F_out, self.F_out)]
+                modules += [torch.Transpose(1, 2)]
+                modules += x
+                modules += [nn.LayerNorm()]
+                modules += [activation_resolver(act, **(act_kwargs or {}))]
+                modules += [Linear(self.F_out, self.F_out)]
+            self.mix.append(Sequential(*modules))
+
+        self.lin = Linear(out_channels, out_channels)
+        self.post_nns = self.mix
+
+        self.reset_parameters()
+
+    def message(self, x_j: Tensor) -> Tensor:
+        return x_j
```

Comment:
I'm not too sure, the GMN should be similar to PNA in its first step, right ? Shouldn't this then  be similar to https://github.com/pyg-team/pytorch_geometric/blob/89f4873b411371880fb64344615ee9babef67d5e/torch_geometric/nn/conv/pna_conv.py#L178-L188 ?